connectors/sources/dropbox.py

```diff
@@ -0,0 +1,173 @@
+
+    def __init__(self, configuration):
+        """Setup the connection to the Dropbox
+
+        Args:
+            configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.dropbox_client = DropboxClient(configuration=configuration)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for Dropbox
+
+        Returns:
+            dictionary: Default configuration.
+        """
+        return {
+            "path": {
+                "label": "Path to fetch files/folders",
+                "order": 1,
+                "required": False,
+                "type": "str",
+                "value": "/",
+                "default_value": "/",
+            },
+            "app_key": {
+                "label": "Dropbox App Key",
+                "sensitive": True,
+                "order": 2,
+                "type": "str",
+                "value": "abc#123",
+            },
+            "app_secret": {
+                "label": "Dropbox App Secret",
+                "sensitive": True,
+                "order": 3,
+                "type": "str",
+                "value": "abc#123",
+            },
+            "refresh_token": {
+                "label": "Dropbox Refresh Token",
+                "sensitive": True,
+                "order": 4,
+                "type": "str",
+                "value": "abc#123",
+            },
+            "include_deleted_files": {
+                "label": "Include Deleted Files",
+                "order": 5,
+                "required": False,
+                "type": "bool",
+                "value": False,
+                "default_value": False,
+            },
+            "retry_count": {
+                "default_value": RETRIES,
+                "display": "numeric",
+                "label": "Maximum retries for failed requests",
+                "order": 6,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": RETRIES,
+                "validations": [{"type": "less_than", "constraint": 10}],
+            },
+        }
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+        Also validate, if user configured path is available in Dropbox."""
+
+        self.configuration.check_valid()
+        await self._remote_validation()
+
+    async def _remote_validation(self):
+        try:
+            if self.dropbox_client.path not in ["", None]:
+                self.dropbox_client.check_path()
+        except BadInputError:
+            raise ConfigurableFieldValueError(
+                "Configured App Key or App Secret is invalid"
+            )
+        except AuthError:
+            raise ConfigurableFieldValueError("Configured Refresh Token is invalid")
+        except ApiError:
```

Comment:
I think is this not the right exception type to validate the path. `ApiError` is a general exception for all API errors. Shouldn't this be `dropbox.exceptions.PathRootError`?

Reply:
You are right, however, if look at the exception object you can see that it breaks down the error. There is a LookupError inside which indicates what exactly went wrong. 

To be more precise in API error handling, we need to handle this particular error. Other ApiError exceptions can go to the general errors.

Reply:
I see, let me make it more precise by catching the `LookupError`.